Sources/SKCore/BuildSystemManager.swift

```diff
@@ -208,11 +205,8 @@ extension BuildSystemManager {
       self.watchedFiles[uri] = (mainFile, language)
     }
 
-    let newStatus = await self.cachedStatusOrRegisterForSettings(for: mainFile, language: language)
-
-    if newStatus.buildSettingsChange != nil, let delegate = self._delegate {
-      await delegate.fileBuildSettingsChanged([uri])
-    }
+    await buildSystem?.registerForChangeNotifications(for: mainFile, language: language)
+    fallbackBuildSystem?.registerForChangeNotifications(for: mainFile, language: language)
```

Comment:
This is now a no-op, should it be removed? Or is there value in keeping it?

Reply:
Oh, good point. I just made `FallbackBuildSystem` not conform to `BuildSystem` and then we can remove `registerForChangeNotifications` there altogether.